src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1269,19 +1278,25 @@ def start(self, id, options):
         Dict(
             'options',
             Bool('force', default=False),
-            Int('timeout', default=90),
         ),
     )
     @job(lock=lambda args: f'stop_vm_{args[0]}_{args[1].get("force") if len(args) == 2 else False}')
     def stop(self, job, id, options):
-        """Stop a VM."""
+        """
+        Stops a VM.
+
+        For unresponsive guests who have exceeded the `shutdown_timeout` defined by the user and have become
+        unresponsive, they required to be powered down using `vm.poweroff`. `vm.stop` is only going to send a
+        shutdown signal to the guest and wait the desired `shutdown_timeout` value before tearing down guest vmemory.
```

Comment:
An option like `force_after_timeout` can be useful, to `poweroff` VM after shutdown timeout, this is exactly what you are doing in `stop_vm`

Reply:
Why do we need to poweroff after timeout? We are already shutting down the machine, arent we?

Reply:
Machine can ignore ACPI power off if it is broken and we need to pull out the power cable forcefully.

Reply:
Why do we care? Its a VM, the host is being powered off

Reply:
It's a `vm.stop` method, it can be called any time, not just on host shutdown